<launch>
<!-- Definicion de variables -->
  <arg name="r1" default="r1"/>
  <arg name="model1" default="waffle"/>
  <arg name="x_pos1" default="0.0"/>
  <arg name="y_pos1" default="0.0"/>
  <arg name="z_pos1" default="0.0"/>
  <arg name="yaw1"   default="0.0"/>

  <arg name="r2" default="r2"/>
  <arg name="model2" default="burger"/>
  <arg name="x_pos2" default="1.3"/>
  <arg name="y_pos2" default="1.3"/>
  <arg name="z_pos2" default="0.0"/>
  <arg name="yaw2"   default="0.5"/>

<!-- Inicia Gazebo con un mundo vacio -->
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find turtlebot3_gazebo)/worlds/empty.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

<!--Grupo1 -->
<group ns = "$(arg r1)">
  <!-- Converte el archivo xacro y lo pone en el servidor de parametros -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model1).urdf.xacro" />
  <!-- Engendra un robot en Gazebo, tomando la descripcion del servidor de parametros -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg r1)" />
    </node>
    
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model1) -x $(arg x_pos1) -y $(arg y_pos1) -z $(arg z_pos1) -Y $(arg yaw1) -param robot_description" />
</group>

<!--Grupo2 -->
<group ns = "$(arg r2)">
  <!-- Converte el archivo xacro y lo pone en el servidor de parametros -->
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model2).urdf.xacro" />
  <!-- Engendra un robot en Gazebo, tomando la descripcion del servidor de parametros -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg r2)" />
    </node>
    
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model2) -x $(arg x_pos2) -y $(arg y_pos2) -z $(arg z_pos2) -Y $(arg yaw2) -param robot_description" />
</group>


<!-- Ejecuta un nodo -->
  <node pkg="uniciclo1" type="uniciclo2_node" name="cualquiera"  output="screen"/>

</launch>
